draft-wang-ppm-dap-taskprov.md

```diff
@@ -56,69 +60,304 @@ collector). Thus it is up to each deployment of DAP to securely implement its
+    /* can be found. */
+    Url aggregator_endpoints<1..2^16-1>; // Defined in I-D.draft-ietf-ppm-dap-02
+
+    /* This determines the query type for batch selection and the */
+    /* properties that all batches for this task must have. */
+    QueryConfig query_config; // Defined in I-D.draft-ietf-ppm-dap-02
+
+    /* The maximum number of times a batch of reports may be queried */
+    /* by the Collector. */
+    uint16 max_batch_lifetime;
+
+    /* Time up to which clients are allowed to upload to this task. */
+    /* See https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/pull/304 */
+    Time task_expiration; // Defined in I-D.draft-ietf-ppm-dap-02
+
+    /* A unique identifier for the VDAF instance used for the task, */
+    /* including the type of measurement associated with the task. */
+    VdafType vdaf_type;
+
+    /* Additional parameters relevant for the vdaf_type */
+    opaque vdaf_data<1..2^16-1>;
+} TaskConfig;
+
+struct {
+    uint32 max_batch_size;
+} FixedSizeQueryConfig;
+
+struct {
+    QueryType query_type; // Defined in I-D.draft-ietf-ppm-dap-02
+    Duration time_precision; // Defined in I-D.draft-ietf-ppm-dap-01
+    uint32 min_batch_size;
+    select (query_type) {
+        case time-interval: Empty;
+        case fixed-size: FixedSizeQueryConfig fixed_size_query_config;
+    }
+} QueryConfig;
 ~~~
 
+The purpose of `TaskConfig` is to include all parameters that are necessary
+for creating a new task in aggregators. It includes all the fields to be
+associated with a task (see task configuration in
+{{?DAP=I-D.draft-ietf-ppm-dap-01}}.). Besides, `TaskConfig` also includes fields
+that useful for configuring a task in-band:
+
+* An opaque `task_info` that is specific to a task. For e.g. this can be a
+string describing the purpose of this task. It can also be the DAP task ID
+chosen by task author.
+
+* An opaque `vdaf_data` that contains any VDAF specific parameters for the
+chosen `vdaf_type`. The aggregators MUST pass `vdaf_data` to VDAF initialiser,
+based on the chosen `vdaf_type`.
+
+The definition of Time, Duration, Url, QueryType follow those in
+{{?DAP=I-D.draft-ietf-ppm-dap-01}}.
+
+Note that the parameters that are not necessarily tied to a task (for e.g.
+collector hpke config), and secrets that should not be known by clients, like
+`vdaf_verify_key`, MUST still be exchanged out-of-band among aggregators.
 
 # Client Behavior
 
 > TODO(wangshan) Say how the Client constructs the extension body. Does the
 > extension change anything else about the upload flow?
 
+The client should know whether `task-prov` extension will be used and all
+parameters required for `TaskConfig` prior to constructing the extension
+body, either out-of-band from aggregators, or from information already saved on
+client.
+
+To use this extension, the client should first decide what is the task ID, see
+{{construct-task-id}}.
+
+## Construct task ID {#construct-task-id}
```

Comment:
yes, but thinking about this more, we probably should enforce task ID to be created involving task config, otherwise the aggregator might aggregate the same task ID with different TaskConfig.